src/extension.js

```diff
@@ -172,6 +172,9 @@ function setExtension( type, name, registrant ){
 
     ext = Renderer;
 
+  } else if (type === '__proto__' || type === 'constructor' || type === 'prototype'){
+    // to avoid potential prototype pollution
+    return util.error( type + ' is an illegal type to be registered, possibly lead to prototype pollutions' );
```

Comment:
Should this validation be handled earlier in the function?

Should the validation be a whitelist instead of a blacklist?  There may be other problematic fields not covered, though you have the big ones.

Should this also be applied to `name`, since there are fields under `prototype` that you may want to protect along the lines of the premise of your PR.